### ISSUE TITLE
[Android] es6 Polyfills

### ISSUE THREAD

**Issue by Thomas101** [facebook/react-native#4841]
Hey,

I've found a few es6 polyfils which behave differently to how they should on android. They work fine on iOS and work fine when you run on android with debug enabled.

``` js
var s = new Set([1,2,3]) // New set created with one item (the array). Should contain 3. Works if you pass in 3 arguments
var a = Array.from(new Set([1])) // creates an empty array rather than [1,2,3]

var m = new Map()
m.set('k', 'v')
var iter = m.keys()
iter.next() // iter.next is undefined. Same for m.values() and m.entries()
```

Tom

**Comment by atom992** [comment]
+1

**Comment by satya164** [comment]
cc @skevy also

**Comment by janicduplessis** [comment]
This probably happens because we use an older version of JSC on android. There's probably a bad or missing polyfill for Map/Set/Iterator

**Comment by bsy** [comment]
+1, any updates on this?

`array.entries()` doesn't work on android, RN 0.27.2

**Comment by satya164** [comment]
@bsy Feel free to send a PR if this is important to you.

**Comment by mkonicek** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with ProductPains! Check out its new home: https://productpains.com/post/react-native/android-es6-polyfills

Product Pains has been very useful in highlighting the top bugs and feature requests:
https://productpains.com/product/react-native?tab=top

Also, if this issue is a bug, **please consider sending a pull request with a fix**.

**Comment by eduardojunio** [comment]
@Thomas101 did you found someway to work-around this issue? I'm having the same problem with TypedArrays, works perfectly on iOS, but not on Android.

**Comment by Thomas101** [comment]
@eduardojunio never got around to fixing it to be honest. Due to time constraints we just avoided using the affected methods. e.g.

```js
const set = new Set()
const arr = [1,2,3]
arr.forEach((v) => set.add(v))
```

rather than `new Set([1,2,3])`

**Comment by eduardojunio** [comment]
@Thomas101 I managed to workaround this by including `import 'core-js'` in the file that is using the typed array (thanks to this issue https://github.com/facebook/react-native/issues/4676#issuecomment-266740462).

**Comment by chirag04** [comment]
i was able to replicate. reopening but feel free to send a PR

**Comment by luco** [comment]
@eduardojunio I couldn't fix it by adding `core-js` on the file. Is there any further step?
My app still crashing on `for` loops.

**Comment by ericketts** [comment]
oh wise stale bot, I petition this issue being re-opened, as it is still occurring for me on react-native 0.53-rc.1 with react 16.2.0. 

Also worth noting this issue is not solved by using core-js (I'm on version 2.5.3), as per this ticket of theirs zloirock/core-js#368

**Comment by jalooc** [comment]
Yes, this issue is still on - shouldn't have been closed. @chrisnojima described the source of this problem really well in https://github.com/zloirock/core-js/issues/368#issuecomment-352294434 (seems to be a problem with babel/core-js). What's worth mentioning: it doesn't happen during remote debugging.

Just a note from my experience with this bug: In my app it's caused when using `viewabilityConfigCallbackPairs` in `FlatList`. Reverting core-js to 2.5.2 didn't help. 

---

One more interesting observation, that may be connected to this issue, is that I get the following warnings *only* when not remote debugging:

* ```
  Warning: React.createElement: type is invalid -- expected ... but got: object
  ```
  ^ with stack trace pointing to `<Fragment>`, where `Fragment` is imported in a legitimate way: 
  `import React, { PureComponent, Fragment } from 'react'` (with v0.53, where fragments' support has already landed);

* ```
  Warning: Each child in an array or iterator should have a unique "key" prop ...
  ```
  ^ this pointed to children of aforementioned `<Fragment>`.

I don't know about it, but seems a bit like other versions of packages where used during remote debugging then during non-remote debugging...
